components/layout_2020/display_list/stacking_context.rs

```diff
@@ -244,7 +244,107 @@ impl StackingContext {
         true
     }
 
-    pub(crate) fn build_display_list<'a>(&self, builder: &'a mut DisplayListBuilder) {
+    /// https://drafts.csswg.org/css-backgrounds/#special-backgrounds
+    ///
+    /// This is only called for the root `StackingContext`
+    pub(crate) fn build_canvas_background_display_list(
+        &self,
+        builder: &mut DisplayListBuilder,
+        fragment_tree: &crate::FragmentTree,
+    ) {
+        let style = if let Some(style) = &fragment_tree.canvas_background.style {
+            style
+        } else {
+            // The root element has `display: none`,
+            // or the canvas background is taken from `<body>` which has `display: none`
+            return;
+        };
+
+        macro_rules! debug_panic {
+            ($msg: expr) => {
+                if cfg!(debug_assertions) {
+                    panic!($msg)
```

Comment:
Maybe emit a warning for release mode so that it's obvious something is wrong?

Reply:
CI runs WPT with debug assertions. Do we expect people who do manual testing to look at a warning log?

Reply:
Emitting a warning in release mode might be interesting if, for instance, this bug became quite common but only for pages on the web. If you feel like this isn't useful though, feel free to ignore this.